vent/io/devices/sensors.py

```diff
@@ -14,76 +15,85 @@ class Sensor(ABC):
 
     def __init__(self):
         """ Upon creation, calls update() to ensure that if get is called there will be something to return."""
-        self._data = np.zeros(
-            self._DEFAULT_STORED_OBSERVATIONS,
-            dtype=np.float16
-        )
-        self._i = 0
-        self._data_length = self._DEFAULT_STORED_OBSERVATIONS
-        self._last_timestamp = -1
+        self._maxlen_data = self._DEFAULT_STORED_OBSERVATIONS
+        self._data = {
+            'timestamp': deque(maxlen=self.maxlen_data),
+            'value': deque(maxlen=self.maxlen_data)
+        }
 
     def update(self) -> float:
         """ Make a sensor reading, verify that it makes sense and store the result internally. Returns True if reading
         was verified and False if something went wrong.
         """
         value = self._read()
-        if self._verify(value):
-            self.__store_last(value)
-            self._last_timestamp = time.time()
+        self._data['timestamp'].append(time.time())
+        self._data['value'].append(value)
         return self._verify(value)
 
-    def get(self) -> float:
-        """ Return the most recent sensor reading."""
-        if self._last_timestamp == -1:
-            raise RuntimeWarning('get() called before update()')
-        return self._data[(self._i - 1) % self._data_length]
+    def get(self, average=False) -> float:

```

Comment:
this method is weird to me - `average=False` will remove an entry from the dqs, but `average=True` doesn't. It also doesn't return timestamp, which I think it probably should, right? how else do timestamps get propagated out of the sensors? want to preserve them rather than having other parts oft he code making timestamps from `time.time()` or smth.